.github/workflows/cibuildwheel.yaml

```diff
@@ -107,35 +112,94 @@ jobs:
           name: cibw-wheel-${{ matrix.python }}-${{ matrix.platform.name }}
           path: wheelhouse/*.whl
 
+  publish-sdist:
+    name: publish-sdist
+    needs:
+      - build-sdist
+      - build-wheels
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 2
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+
+      - name: Download artifacts
+        uses: actions/download-artifact@v4
+        with:
+          pattern: cibw-sdist
+          path: dist
+          merge-multiple: true
+
+      - name: Check collected artifacts
+        run: |
+          tools/check_dist.py ${{ github.event_name == 'release' && format('-r {0}', github.ref) || '' }}
```

Comment:
You run this twice. I don't think we need to

Reply:
I thought it wouldn't hurt, at least it is nice to check it here verifying we have downloaded the correct thing using the action.
I can remove it from the `sdist` step if you think it is redundant?

Reply:
Maybe it's fine to keep. It's just a fancy ls